html/modules/custom/bc_dc/src/Controller/BcDcCreateFileController.php

```diff
@@ -0,0 +1,111 @@
+      }
+      $sheetData= [
+        ['column_allowed_values',
+          'column_data_quality',
+          'column_description',
+          'column_name',
+          'column_size',
+          'column_transformations',
+          'provenance_field_name',
+          'provenance_field_number',
+          'provenance_field_question',
+          'provenance_form_name',
+          'provenance_form_number',
+        ],
+        [$paragraph->get('field_column_allowed_values')->value,
+          $paragraph->get('field_column_data_quality')->value,
+          $paragraph->get('field_column_description')->value,
+          $paragraph->get('field_column_name')->value,
+          $paragraph->get('field_column_size')->value,
+          $paragraph->get('field_column_transformations')->value,
+          $paragraph->get('field_provenance_field_name')->value,
+          $paragraph->get('field_provenance_field_number')->value,
+          $paragraph->get('field_provenance_field_question')->value,
+          $paragraph->get('field_provenance_form_name')->value,
+          $paragraph->get('field_provenance_form_number')->value,
+          ]
+      ];
+
+      $spreadsheet = new Spreadsheet();
+
+      if ($param === 'xlsx') {
+        $filename = $path . '_ID_' . $nid . '.xlsx';
+        // Save to file.
+        $worksheets = new \PhpOffice\PhpSpreadsheet\Worksheet\Worksheet($spreadsheet, "Sheet 1");
+        $spreadsheet->addSheet($worksheets, 0);
+        $worksheets->fromArray($sheetData);
+        foreach ($worksheets as $worksheet) {
+          foreach ($worksheet->getColumnIterator() as $column) {
+            $worksheet->getColumnDimension($column->getColumnIndex())->setAutoSize(true);
+          }
+        }
+        $writer = new \PhpOffice\PhpSpreadsheet\Writer\Xlsx($spreadsheet);
+        $directory = 'public://data_dictionary/' . $filename;
+        $writer->save($directory);
+
+      }
+      elseif ($param === 'csv') {
+        $filename = $path . '_ID_' . $nid . '.csv';
+        $mySpreadsheet = new Spreadsheet();
+        $worksheets = new \PhpOffice\PhpSpreadsheet\Worksheet\Worksheet($mySpreadsheet, "Sheet 1");
+        $mySpreadsheet->addSheet($worksheets, 0);
+        $worksheets->fromArray($sheetData);
+        foreach ($worksheets as $worksheet) {
+          foreach ($worksheet->getColumnIterator() as $column) {
+            $worksheet->getColumnDimension($column->getColumnIndex())->setAutoSize(TRUE);
+          }
+        }
+        $writer = new \PhpOffice\PhpSpreadsheet\Writer\Csv($mySpreadsheet);
+        $directory = 'public://data_dictionary/' . $filename;
+        $writer->setDelimiter(',');
+        $writer->setEnclosure('');
+        $writer->setLineEnding("\r\n");
+        $writer->setSheetIndex(0);
+        $writer->save($directory);
+
+      }
+
+    }
+    else {
+      throw new NotFoundHttpException();
+    }
+
+    $uri = $directory;
```

Comment:
Why is there both `$uri` and `$directory`? They seem to be the same thing.

Reply:
fixed